tests/std/tests/P0896R4_counted_iterator/test.cpp

```diff
@@ -302,6 +302,40 @@ struct instantiator {
     }
 };
 
+// Also test P2259R1 Repairing input range adaptors and counted_iterator
+struct simple_input_iter {
+    using value_type        = double;
+    using difference_type   = long;
+    using iterator_category = input_iterator_tag;
+    using iterator_concept  = forward_iterator_tag;
+
+    value_type operator*() const;
+    simple_input_iter& operator++();
+    simple_input_iter operator++(int);
+
+    bool operator==(simple_input_iter const&) const;
+};
+
+using CI = counted_iterator<simple_input_iter>;
+
+static_assert(same_as<iterator_traits<simple_input_iter>::iterator_category, input_iterator_tag>);
+static_assert(forward_iterator<simple_input_iter>);
+static_assert(forward_iterator<CI>);
+static_assert(!contiguous_iterator<CI>);
+static_assert(same_as<CI::value_type, double>);
+static_assert(same_as<CI::iterator_category, input_iterator_tag>);
+static_assert(same_as<CI::iterator_concept, forward_iterator_tag>);

```

Comment:
Could we have another test case wrapping an iterator with no nested typenames that specializes `iterator_traits` directly?

Reply:
I think this is covered by the existing tests (that use `test::iterator`).

Reply:
Yes, that's sufficient.